### PR TITLE
[EC-137] Static analysis tool fixes

### PR DESCRIPTION
**Pull request by ntallar**
## Description

Includes several fixes to issues detected while running different static analysis tools (https://github.com/HairyFotr/linter, https://github.com/sksamuel/scapegoat and http://www.wartremover.org/).
The main problems fixed were:
- Changed `Seq()` to `Seq.empty`.
- Removed unnecessary code such as unused method and class parameters.
- Parantheless methods with effect and returning unit.
- Closed `Source` and did it inside a `finally` clause.

### CODE REVIEW

**Comment by ntallar** [comment]
As a week has passed since the last time the static analysis tools have been run, I will update this PR with the required changes for the latest version of the code.

**Comment by LukasGasior1** [comment]
how about `Nil` instead of `Seq.empty`?

File: src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncNodesRequestHandler.scala [hunk 1]
```diff
@@ -31,20 +31,20 @@ class FastSyncNodesRequestHandler(
...
     val hashesToRequest = (nodeData.values.indices zip receivedHashes) flatMap { case (idx, valueHash) =>
       requestedHashes.find(_.v == valueHash) map {
-        case StateMptNodeHash(hash) =>
-          handleMptNode(hash, nodeData.getMptNode(idx))
+        case StateMptNodeHash(_) =>
+          handleMptNode(nodeData.getMptNode(idx))
 
-        case ContractStorageMptNodeHash(hash) =>
-          handleContractMptNode(hash, nodeData.getMptNode(idx))
+        case ContractStorageMptNodeHash(_) =>
```
Comment (LukasGasior1) [inline src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncNodesRequestHandler.scala]: generally I prefer to use `_: ContractStorageMptNodeHash` that way if at some point someone adds new field to `ContractStorageMptNodeHash` this match will still compile

File: src/main/scala/io/iohk/ethereum/crypto/EthereumIESEngine.scala [hunk 2]
```diff
@@ -131,8 +131,7 @@ class EthereumIESEngine(kdf: Either[ConcatKDFBytesGenerator, MGF1BytesGeneratorE
     IV.foreach(iv => mac.update(iv, 0, iv.length))
     mac.update(cryptogram, inOff + encodedPublicKey.length, inLen - encodedPublicKey.length - messageAuthenticationCodeCalculated.length)
 
-    if (macData.isDefined)
-      mac.update(macData.get, 0, macData.get.length)
+    macData foreach { data => mac.update(data, 0, data.length) }
```
Comment (LukasGasior1) [inline src/main/scala/io/iohk/ethereum/crypto/EthereumIESEngine.scala]: :+1:

File: src/main/scala/io/iohk/ethereum/vm/OpCode.scala [hunk 3]
```diff
@@ -617,7 +617,7 @@ case object LOG3 extends LogOp(0xa3)
 case object LOG4 extends LogOp(0xa4)
 
 
-sealed abstract class CallOp(code: Int, delta: Int, alpha: Int) extends OpCode(code, delta, 1, G_zero) {
+sealed abstract class CallOp(code: Int, delta: Int, alpha: Int) extends OpCode(code, delta, alpha, G_zero) {
```
Comment (LukasGasior1) [inline src/main/scala/io/iohk/ethereum/vm/OpCode.scala]: nice catch

**Comment by LukasGasior1** [comment]
other than my two minor comments, LGTM!

**Comment by AlanVerbner** [comment]
@ntallar you need to update :(

**Pull request merged**